Sources/XiEditor/Core/Types.swift

```diff
@@ -0,0 +1,115 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+enum UpdateOperationType: String {
+	typealias RawValue = String
+
+	case Copy = "copy"
+	case Invalidate = "invalidate"
+	case Insert = "ins"
+	case Update = "update"
+	case Skip = "skip"
```

Comment:
The coding style of the enum case should be lower camel case. In Swift guideline and other companies' coding style, enum case is lower camel case.
https://docs.swift.org/swift-book/LanguageGuide/Enumerations.html